Description: Ensure member_id is correctly encoded
Author: James Page <james.page@ubuntu.com>
Forwarded: no

--- a/gnocchi/cli/metricd.py
+++ b/gnocchi/cli/metricd.py
@@ -60,12 +60,14 @@ class MetricProcessBase(cotyledon.Servic
         self._wake_up.set()
 
     def _configure(self):
-        member_id = "%s.%s.%s" % (socket.gethostname(),
-                                  self.worker_id,
-                                  # NOTE(jd) Still use a uuid here so we're
-                                  # sure there's no conflict in case of
-                                  # crash/restart
-                                  str(uuid.uuid4()))
+        member_id = str.encode(
+            "%s.%s.%s" % (socket.gethostname(),
+                          self.worker_id,
+                          # NOTE(jd) Still use a uuid here so we're
+                          # sure there's no conflict in case of
+                          # crash/restart
+                          str(uuid.uuid4()))
+        )
         self.coord = get_coordinator_and_start(member_id,
                                                self.conf.coordination_url)
         self.store = storage.get_driver(self.conf, self.coord)
--- a/gnocchi/rest/app.py
+++ b/gnocchi/rest/app.py
@@ -93,7 +93,7 @@ class GnocchiHook(pecan.hooks.PecanHook)
                         # entirely.
                         self.backends[name] = (
                             metricd.get_coordinator_and_start(
-                                str(uuid.uuid4()),
+                                str.encode(str(uuid.uuid4())),
                                 self.conf.coordination_url)
                         )
                     elif name == "storage":
